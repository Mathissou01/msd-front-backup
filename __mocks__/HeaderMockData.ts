import { GetContractMenuDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetContractMenuDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contract: {
          data: {
            attributes: {
              contractMenu: {
                data: {
                  attributes: {
                    serviceLinks: [
                      {
                        name: "Carte",
                        isDisplayed: true,
                        picto: {
                          data: {
                            attributes: {
                              url: "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-grymas/address.svg",
                            },
                          },
                        },
                      },
                      {
                        name: "Collecte à mon adresse",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Guide du tri",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Faire une demande",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                      },
                      {
                        name: "Réduire mes déchets",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                        freeContentSubService: {
                          data: {
                            id: "2",
                            attributes: {
                              name: "Réduire mes déchets",
                            },
                          },
                        },
                      },
                      {
                        name: "Valoriser mes déchets",
                        isDisplayed: true,
                        picto: {
                          data: null,
                        },
                        freeContentSubService: {
                          data: {
                            id: "1",
                            attributes: {
                              name: "Valoriser mes déchets",
                            },
                          },
                        },
                      },
                    ],
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
