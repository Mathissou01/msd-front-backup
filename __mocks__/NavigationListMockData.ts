import { GetContractMenuDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetContractMenuDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contract: {
          __typename: "ContractEntityResponse",
          data: {
            __typename: "ContractEntity",
            attributes: {
              __typename: "Contract",
              contractMenu: {
                __typename: "ContractMenuEntityResponse",
                data: {
                  __typename: "ContractMenuEntity",
                  attributes: {
                    __typename: "ContractMenu",
                    serviceLinks: [
                      {
                        __typename: "ComponentLinksMap",
                        name: "Carte",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: {
                            __typename: "UploadFileEntity",
                            attributes: {
                              __typename: "UploadFile",
                              url: "https://fake-url/address.svg",
                            },
                          },
                        },
                      },
                      {
                        __typename: "ComponentLinksCalendar",
                        name: "Collecte à mon adresse",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                      },
                      {
                        __typename: "ComponentLinksRecycling",
                        name: "Guide du tri",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                      },
                      {
                        __typename: "ComponentLinksRequest",
                        name: "Faire une demande",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                      },
                      {
                        __typename: "ComponentLinksFrees",
                        name: "Réduire mes déchets",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                        freeContentSubService: {
                          __typename: "FreeContentSubServiceEntityResponse",
                          data: {
                            __typename: "FreeContentSubServiceEntity",
                            id: "2",
                            attributes: {
                              __typename: "FreeContentSubService",
                              name: "Réduire mes déchets",
                            },
                          },
                        },
                      },
                      {
                        __typename: "ComponentLinksFrees",
                        name: "Valoriser mes déchets",
                        isDisplayed: true,
                        picto: {
                          __typename: "UploadFileEntityResponse",
                          data: null,
                        },
                        freeContentSubService: {
                          __typename: "FreeContentSubServiceEntityResponse",
                          data: {
                            __typename: "FreeContentSubServiceEntity",
                            id: "1",
                            attributes: {
                              __typename: "FreeContentSubService",
                              name: "Valoriser mes déchets",
                            },
                          },
                        },
                      },
                    ],
                  },
                },
              },
            },
          },
        },
      },
    },
  },
];
