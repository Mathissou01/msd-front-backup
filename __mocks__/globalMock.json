{
  "contract": {
    "id": "1",
    "attributes": {
      "clientName": "Grand Test",
      "clientType": "city",
      "contractStatus": "Initialisation",
      "isNonExclusive": false,
      "isRVFrance": true,
      "logo": {
        "data": {
          "id": "26",
          "attributes": {
            "name": "images.png",
            "mime": "image/png",
            "size": 2.24,
            "url": "https://stomsdmediadev.blob.core.windows.net/msd-media-dev/assets-local-root/images.png",
            "provider": "azure-storage-blob",
            "hash": "images_eb8049ec37",
            "alternativeText": null
          }
        }
      },
      "contractCustomization": {
        "data": {
          "id": "2",
          "attributes": {
            "footer": {
              "data": {
                "attributes": {
                  "accessibilityLevel": null,
                  "cguSubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "CGU",
                        "link": null
                      }
                    }
                  },
                  "accessibilitySubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Accessibilité",
                        "link": null
                      }
                    }
                  },
                  "confidentialitySubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Confidentialité",
                        "link": null
                      }
                    }
                  },
                  "cookiesSubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Cookies",
                        "link": null
                      }
                    }
                  },
                  "contactUsSubService": {
                    "data": {
                      "id": "1",
                      "attributes": {
                        "name": "Contactez-nous",
                        "label": "Contactez-nous",
                        "link": null,
                        "isActivated": false
                      }
                    }
                  }
                }
              }
            }
          }
        }
      },
      "contractMenu": {
        "data": {
          "attributes": {
            "serviceLinks": [
              {
                "__typename": "ComponentLinksRecyclingGuide",
                "id": "3",
                "name": "Guide du tri",
                "isDisplayed": true,
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksNews",
                "id": "2",
                "name": "Actualités",
                "isDisplayed": true,
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksEvents",
                "id": "2",
                "name": "Évenements",
                "isDisplayed": true,
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksFrees",
                "id": "3",
                "name": "Editorial 1",
                "isDisplayed": true,
                "freeContents": {
                  "data": [
                    {
                      "id": "1",
                      "attributes": {
                        "name": "Editorial 1"
                      }
                    }
                  ]
                },
                "picto": {
                  "data": null
                }
              },
              {
                "__typename": "ComponentLinksFrees",
                "id": "4",
                "name": "Editorial 2",
                "isDisplayed": true,
                "freeContents": {
                  "data": [
                    {
                      "id": "2",
                      "attributes": {
                        "name": "Editorial 2"
                      }
                    }
                  ]
                },
                "picto": {
                  "data": null
                }
              }
            ]
          }
        }
      }
    }
  },
  "colors": {
    "external-contrast-text": "#FFFFFF",
    "external-primary": "#0A959F",
    "calculated-primary-dark": "#006771",
    "calculated-primary-light": "#0a959f1a",
    "external-secondary": "#A3C037",
    "calculated-secondary-dark": "#719000",
    "calculated-secondary-light": "#a3c0371a",
    "external-waste-household": "#6C757D",
    "external-waste-selective": "#FFC229",
    "external-waste-bio": "#BD4602",
    "external-waste-green": "#9BCD41",
    "external-waste-glass": "#30A23D",
    "external-waste-paper": "#2068C7",
    "external-waste-other": "#FFFFFF"
  }
}
