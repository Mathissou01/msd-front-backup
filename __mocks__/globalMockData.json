{
  "contractMenu": {
    "data": {
      "attributes": {
        "serviceLinks": [
          {
            "__typename": "ComponentLinksMap",
            "name": "Carte",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksCalendar",
            "name": "Collecte à mon adresse",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksRequest",
            "name": "Faire une demande",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksRecycling",
            "name": "Guide du tri",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksFrees",
            "name": "Réduire mes déchets",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          },
          {
            "__typename": "ComponentLinksFrees",
            "name": "Valoriser mes déchets",
            "isDisplayed": true,
            "picto": {
              "data": null
            }
          }
        ]
      }
    }
  }
}
