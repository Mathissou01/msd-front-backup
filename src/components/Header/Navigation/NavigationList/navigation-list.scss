@use "src/styles/01-settings/settings.colors" as colors;

.c-NavigationList {
  $self: &;

  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 0;
  list-style: none;

  &_desktop {
    flex-grow: 1;
    align-items: flex-start;
  }

  &__Item {
    display: flex;
    flex-direction: column;
    width: 100%;

    &:hover {
      background-color: colors.$brand-primary-light;
    }

    #{$self}_mobile & {
      height: 66px;
    }

    #{$self}_desktop & {
      flex-grow: 1;
      gap: 6px;
      align-items: center; // TODO: remove after replacing all items with NavigationListButton
      justify-content: center;
      min-height: 70px;
      max-height: 102px;
      border-bottom: 1px solid colors.$gray-200;
    }

    &:first-child:not(&_active) {
      border-top: 1px solid colors.$gray-200;
    }

    &_active {
      #{$self}_desktop & {
        background-color: colors.$brand-primary;
      }
    }
  }
}
