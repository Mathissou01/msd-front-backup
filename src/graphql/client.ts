import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache(),
  });
}

export default client;
