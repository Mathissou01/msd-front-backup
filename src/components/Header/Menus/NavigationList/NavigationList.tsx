import classNames from "classnames";
import React from "react";
import {
  ENavigationPages,
  useNavigation,
} from "../../../../hooks/useNavigation";
import NavigationListButton from "./NavigationListButton/NavigationListButton";
import { useGetContractMenuQuery } from "../../../../graphql/codegen/generated-types";
import { normalizeStringPath, removeNulls } from "../../../../lib/utilities";
import CommonSpinner from "../../../Common/CommonSpinner/CommonSpinner";
import "./navigation-list.scss";

export interface INavigationLink {
  type?: string;
  name: string;
  isDisplayed: boolean;
  picto?: { data?: { attributes?: { url: string } } };
  path?: string;
}

// eslint-disable-next-line
export function isNavigationLink(link: any): link is INavigationLink {
  return "name" in link && "isDisplayed" in link;
}

interface INavigationListProps {
  isDesktopMode: boolean;
  onNavigationClick?: () => void;
}

export default function NavigationList({
  isDesktopMode,
  onNavigationClick,
}: INavigationListProps) {
  const homepageLabel = "Accueil";
  const defaultPicto = "/images/pictos/default.svg";

  /* Methods */
  function isActiveRoute(route: ENavigationPages | string) {
    return route === currentPage ? "c-NavigationList__Item_active" : "";
  }

  function handleClick(path: string) {
    setCurrentPage(path);
    if (onNavigationClick) {
      onNavigationClick();
    }
  }

  /* External Data */
  const contractId = "1";
  const { loading, error, data } = useGetContractMenuQuery({
    variables: { contractId },
  });
  const { currentPage, setCurrentPage } = useNavigation();

  const contractMenus: Array<INavigationLink> | null =
    data?.contract?.data?.attributes?.contractMenu?.data?.attributes?.serviceLinks
      ?.map((link) => {
        if (link) {
          const type = link.__typename;
          let path = "/";
          if (type === "ComponentLinksFrees" && link.name) {
            path = normalizeStringPath(link.name);
          }
          if (type && isNavigationLink(link)) {
            return {
              type,
              name: link?.name,
              isDisplayed: link?.isDisplayed,
              picto: link?.picto,
              path,
            };
          }
        }
      })
      .filter(removeNulls) ?? null;

  /* Local Data */
  const dynamicClassNames = classNames("c-NavigationList", {
    "c-NavigationList_desktop": isDesktopMode,
    "c-NavigationList_mobile": !isDesktopMode,
  });

  if (loading) return <CommonSpinner />;
  if (error) return <span>{error?.message}</span>;
  return (
    <ul className={dynamicClassNames}>
      <li
        className={`c-NavigationList__Item ${isActiveRoute(
          ENavigationPages["HomeLink"],
        )}`}
      >
        <NavigationListButton
          href={ENavigationPages["HomeLink"]}
          label={homepageLabel}
          pictoUrl="/images/pictos/homepage.svg"
          isDesktopMode={isDesktopMode}
          isActive={currentPage === ENavigationPages["HomeLink"]}
          onClick={() => handleClick(ENavigationPages["HomeLink"])}
        />
      </li>
      {contractMenus &&
        contractMenus.map((menu) => {
          if (
            menu?.isDisplayed &&
            menu?.type &&
            menu?.type in ENavigationPages
          ) {
            let path: string =
              Object.values(ENavigationPages)[
                Object.keys(ENavigationPages).indexOf(menu.type)
              ];
            if (menu.type === "ComponentLinksFrees") {
              path += `/${menu.path}`;
            }
            return (
              <li
                key={`navigation_link_${path}`}
                className={`c-NavigationList__Item ${isActiveRoute(path)}`}
              >
                <NavigationListButton
                  href={path}
                  label={menu.name}
                  pictoUrl={menu.picto?.data?.attributes?.url ?? defaultPicto}
                  isDesktopMode={isDesktopMode}
                  isActive={currentPage === path}
                  onClick={() => handleClick(path)}
                />
              </li>
            );
          }
        })}
    </ul>
  );
}
