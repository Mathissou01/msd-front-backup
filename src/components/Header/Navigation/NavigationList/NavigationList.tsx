import classNames from "classnames";
import React from "react";
import {
  IServiceLink,
  remapServiceLinksDynamicZone,
} from "../../../../lib/service-links";
import {
  ENavigationPages,
  useNavigation,
} from "../../../../hooks/useNavigation";
import { useContract } from "../../../../hooks/useContract";
import NavigationListButton from "./NavigationListButton/NavigationListButton";
import "./navigation-list.scss";
import { removeNulls } from "../../../../lib/utilities";

interface INavigationListProps {
  isDesktopMode: boolean;
  onNavigationClick?: () => void;
}

export default function NavigationList({
  isDesktopMode,
  onNavigationClick,
}: INavigationListProps) {
  /* Static Data */
  const homepageLabel = "Accueil";
  const defaultPicto = "/images/pictos/default.svg";

  /* Methods */
  function isActiveRoute(route: ENavigationPages | string) {
    return route === currentPage ? "c-NavigationList__Item_active" : "";
  }

  function handleClick(path: string) {
    setCurrentPage(path);
    if (onNavigationClick) {
      onNavigationClick();
    }
  }

  /* External Data */
  const { contract } = useContract();
  const { currentPage, setCurrentPage } = useNavigation();

  const contractMenus: Array<IServiceLink> | null =
    remapServiceLinksDynamicZone(
      contract.attributes?.contractMenu?.data?.attributes?.serviceLinks?.filter(
        removeNulls,
      ) ?? [],
    );

  /* Local Data */
  const dynamicClassNames = classNames("c-NavigationList", {
    "c-NavigationList_desktop": isDesktopMode,
    "c-NavigationList_mobile": !isDesktopMode,
  });

  return (
    <ul className={dynamicClassNames}>
      <li
        className={`c-NavigationList__Item ${isActiveRoute(
          ENavigationPages["HomeLink"],
        )}`}
      >
        <NavigationListButton
          href={ENavigationPages["HomeLink"]}
          label={homepageLabel}
          pictoUrl="/images/pictos/homepage.svg"
          isDesktopMode={isDesktopMode}
          isActive={currentPage === ENavigationPages["HomeLink"]}
          onClick={() => handleClick(ENavigationPages["HomeLink"])}
        />
      </li>
      {/* <li
        className={`c-NavigationList__Item ${isActiveRoute(
          ENavigationPages["MonCompteurDechets"],
        )}`}
      >
        <NavigationListButton
          href={ENavigationPages["MonCompteurDechets"]}
          label="mon compteur dÃ©chets"
          pictoUrl="/images/pictos/mon_compteur_dechets.svg"
          isDesktopMode={isDesktopMode}
          isActive={currentPage === ENavigationPages["MonCompteurDechets"]}
          onClick={() => handleClick(ENavigationPages["MonCompteurDechets"])}
        />
      </li> */}
      {contractMenus &&
        contractMenus.map((menu, index) => {
          if (
            menu?.isDisplayed &&
            menu?.type &&
            menu?.type in ENavigationPages
          ) {
            return (
              <li
                key={`navigation_link_${index}`}
                className={`c-NavigationList__Item ${isActiveRoute(
                  menu.generatedSlug,
                )}`}
              >
                <NavigationListButton
                  href={menu.generatedSlug}
                  label={menu.name}
                  pictoUrl={menu.picto?.data?.attributes?.url ?? defaultPicto}
                  isDesktopMode={isDesktopMode}
                  isActive={currentPage === menu.generatedSlug}
                  onClick={() => handleClick(menu.generatedSlug)}
                />
              </li>
            );
          }
        })}
    </ul>
  );
}
