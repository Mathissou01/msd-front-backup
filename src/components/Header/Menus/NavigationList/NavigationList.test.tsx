import { MockedProvider } from "@apollo/client/testing";
import { render, screen } from "@testing-library/react";
import { defaultMockData } from "../../../../../__mocks__/NavigationListMockData";
import NavigationList from "./NavigationList";

describe("NavigationList", () => {
  it("renders in mobile mode", async () => {
    const { container } = render(
      <MockedProvider mocks={defaultMockData}>
        <NavigationList isDesktopMode={false} />
      </MockedProvider>,
    );

    expect(await screen.findByTestId("common-spinner")).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
    expect(await screen.findByText("Accueil")).toBeInTheDocument();
    expect(
      await screen.findByText("Collecte Ã  mon adresse"),
    ).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
  });
});
