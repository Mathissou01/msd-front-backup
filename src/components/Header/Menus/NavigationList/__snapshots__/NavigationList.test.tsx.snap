// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationList renders in mobile mode 1`] = `
<div>
  <div
    class="c-CommonSpinner__Wrapper"
  >
    <div
      class="c-CommonSpinner"
      data-testid="common-spinner"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`NavigationList renders in mobile mode 2`] = `
<div>
  <ul
    class="c-NavigationList c-NavigationList_mobile"
  >
    <li
      class="c-NavigationList__Item c-NavigationList__Item_active"
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile c-NavigationListButton_active"
        href="/"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="/images/pictos/homepage.svg"
              srcset="/images/pictos/homepage.svg 1x, /images/pictos/homepage.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Accueil
        </span>
      </a>
    </li>
    <li
      class="c-NavigationList__Item "
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile"
        href="/service/carte"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="https://fake-url/address.svg"
              srcset="https://fake-url/address.svg 1x, https://fake-url/address.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Carte
        </span>
      </a>
    </li>
    <li
      class="c-NavigationList__Item "
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile"
        href="/service/collecte-adresse"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="/images/pictos/default.svg"
              srcset="/images/pictos/default.svg 1x, /images/pictos/default.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Collecte à mon adresse
        </span>
      </a>
    </li>
    <li
      class="c-NavigationList__Item "
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile"
        href="/service/guide-tri"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="/images/pictos/default.svg"
              srcset="/images/pictos/default.svg 1x, /images/pictos/default.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Guide du tri
        </span>
      </a>
    </li>
    <li
      class="c-NavigationList__Item "
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile"
        href="/service/demandes"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="/images/pictos/default.svg"
              srcset="/images/pictos/default.svg 1x, /images/pictos/default.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Faire une demande
        </span>
      </a>
    </li>
    <li
      class="c-NavigationList__Item "
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile"
        href="/edito/reduire-mes-dechets"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="/images/pictos/default.svg"
              srcset="/images/pictos/default.svg 1x, /images/pictos/default.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Réduire mes déchets
        </span>
      </a>
    </li>
    <li
      class="c-NavigationList__Item "
    >
      <a
        class="c-NavigationListButton c-NavigationListButton_mobile"
        href="/edito/valoriser-mes-dechets"
      >
        <div
          class="c-NavigationListButton__Circle"
        >
          <div
            class="c-NavigationListButton__Picto"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="24"
              loading="lazy"
              src="/images/pictos/default.svg"
              srcset="/images/pictos/default.svg 1x, /images/pictos/default.svg 2x"
              style="color: transparent;"
              width="24"
            />
          </div>
        </div>
        <span
          class="c-NavigationListButton__Label"
        >
          Valoriser mes déchets
        </span>
      </a>
    </li>
  </ul>
</div>
`;
